### PR TITLE
[WFLY-19683][WFLY-19730] Provide SSLContext via the new ClientCustomizers

### PR DESCRIPTION
**Pull request by kabir**
http://issues.redhat.com/browse/WFLY-19683

https://issues.redhat.com/browse/WFLY-19730

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19683](https://issues.redhat.com/browse/WFLY-19683)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -443,7 +443,8 @@
         <version.io.smallrye.smallrye-opentelemetry>2.6.0</version.io.smallrye.smallrye-opentelemetry>
         <version.io.smallrye.smallrye-reactive-messaging>4.24.0</version.io.smallrye.smallrye-reactive-messaging>
         <version.io.undertow.jastow>2.2.8.Final</version.io.undertow.jastow>
-        <version.io.vertx.vertx>4.5.9</version.io.vertx.vertx>
+        <version.io.vertx.vertx>4.5.10
+        </version.io.vertx.vertx>
```
Comment (rhusar) [inline pom.xml:447]: BTW accidental extraneous new line here.

**Comment by kabir** [comment]
@bstansberry this finally passed CI. I pushed a fix for Rado's comment

File: microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/WildFlyKafkaSSLEngineFactory.java [hunk 2]
```diff
@@ -20,18 +17,22 @@
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.SSLParameters;
 
+import org.wildfly.microprofile.reactive.messaging.common.security.ElytronSSLContextRegistry;
+
 /**
  * @author <a href="mailto:kabir.khan@jboss.com">Kabir Khan</a>
  */
 public class WildFlyKafkaSSLEngineFactory implements org.apache.kafka.common.security.auth.SslEngineFactory {
+    static final String SSL_ENGINE_FACTORY_CLASS = "ssl.engine.factory.class";
```
Comment (bstansberry) [inline microprofile/reactive-messaging-smallrye/kafka/src/main/java/org/wildfly/microprofile/reactive/messaging/config/kafka/ssl/context/WildFlyKafkaSSLEngineFactory.java:26]: Nit for future consideration: This constant is declared here but isn't used here. AFAICT it's only used in a different class.

File: pom.xml [hunk 3]
```diff
@@ -444,7 +444,7 @@
         <version.io.smallrye.smallrye-opentelemetry>2.6.0</version.io.smallrye.smallrye-opentelemetry>
         <version.io.smallrye.smallrye-reactive-messaging>4.24.0</version.io.smallrye.smallrye-reactive-messaging>
         <version.io.undertow.jastow>2.2.8.Final</version.io.undertow.jastow>
-        <version.io.vertx.vertx>4.5.9</version.io.vertx.vertx>
+        <version.io.vertx.vertx>4.5.10</version.io.vertx.vertx>
```
Comment (bstansberry) [inline pom.xml:447]: I filed https://issues.redhat.com/browse/WFLY-19730 and will link this to it and update the PR title so the JIRA appears in the git log. If you need to update this PR for another reason, please amend the commit message.

Also, please add the diff or rn stuff to the WFLY-19730 description. If it resolves any CVEs please change the JIRA title to note that.

**Comment by bstansberry** [comment]
Thanks @kabir and @rhusar

**Pull request merged**